src/cluster_slots.c

```diff
@@ -0,0 +1,217 @@
+/* Cluster slots APIs and commands - to retrieve, update and process slot level data 
```

Comment:
I remember we named this because we planned to break up the APIs, but we ended up naming it weirdly (what is a "legacy cluster" when there is only one cluster) but it doesn't make a lot of sense to revert it now. I don't mind this in its own file, but maybe we call it cluster_slot_stats instead?